mining.md

```diff
@@ -42,24 +42,18 @@ TODO: sectors need to be globally unique. This can be done either by having the
 
 #### Step 2: Proving Storage (PoSt creation)
 
-At the beginning of their proving period, miners collect the proving set (the set of all live sealed sectors on the chain at this point), and then call `ProveStorage`. This process will take the entire proving period to complete.
-
 ```go
-func ProveStorage(sectorSize BytesAmount, sectors []commR, startTime BlockHeight) (PoSTProof, []FaultSet) {
-    var proofs []Proofs
-    var seeds []Seed
-    var faults []FaultSet
-    for t := 0; t < ProvingPeriod; t += ReseedPeriod {
-        seeds = append(seeds, GetSeedFromBlock(startTime+t))
-        proof, faultset := GenPost(sectors, seeds[t], vdfParams)
-        proofs = append(proofs, proof)
-        faults = append(faults, faultset)
-    }
-    return GenPostSnark(sectorSize, sectors, seeds, proofs), faults
+func ProveStorage(sectorSize BytesAmount, sectors []commR) PoStProof {
+    challengeBlockHeight := miner.ProvingPeriodStart + ProvingPeriodDuratoin(sectorSize) - POST_CHALLENGE_TIME
+
+    // Faults to be used are the currentFaultSet for the miner.
+    faults := miner.currentFaultSet
+    seed := GetRandFromBlock(challengeBlockHeight)
```

Comment:
As stated in the other comment, this is incorrect. The randomness must come from exactly `challengeBlockHeight`, no lookback.

Reply:
fixed